### PR TITLE
fix: Change map to geo_equals

### LINKED ISSUE #2197: mutate geo column causes TypeError: geo_equals() got an unexpected keyword argument 'cache'
Using `mutate` with the same GEO column more than once causes an error.  The error happens when printing or using `compile()`.  If I only reference the same column once, the result is correct.

I'm using this with omniscidb.

The `geo_equals` function does not have a cache arg: https://github.com/ibis-project/ibis/blob/cc0ebb6242e75670d282476d6ea10bf5b8583fc4/ibis/expr/api.py#L1845

```
tm = t.mutate([
    t['Location'].y().name('Location_Latitude'),
    t['Location'].y().name('Latitude')])
tm.compile()
```

```
/opt/conda/lib/python3.7/site-packages/ibis/expr/operations.py in all_equal(left, right, cache)
    171 
    172     if hasattr(left, 'equals'):
--> 173         return left.equals(right, cache=cache)
    174     return left == right
    175 

TypeError: geo_equals() got an unexpected keyword argument 'cache'
```

### PR DESCRIPTION
**Pull request by xmnlab**
This PR fix the problem with `geo_equals`.

Currently, `equals` operation is mapped to `geo_equals` and it is given problems as reported at #2197

Fix #2197

### CODE REVIEW

File: ibis/backends/tests/test_geospatial.py [hunk 1]
```diff
@@ -686,3 +686,45 @@ def test_geo_ops_smoke(backend, fn_expr):
     """Smoke tests for geo spatial operations."""
     geo_table = backend.geo
     assert fn_expr(geo_table).compile() != ''
+
+
+@pytest.mark.only_on_backends(["postgres"])
+@pytest.mark.xfail_unsupported
+def test_geo_equals(backend, geo):
+    """
+    Test geo_equals/equals.
```
Comment (jreback) [inline ibis/backends/tests/test_geospatial.py:695]: reference the issue in a comment. the Notes section is not needed here

**Review by jreback — CHANGES_REQUESTED** [review]
tests are failing; also pls add a release note

**Review by jreback — APPROVED** [review]

**Comment by xmnlab** [comment]
@jreback thanks for the review.

I added the release note and also changed 2 tests that were using `.equals`. to `geo_equals`.
In ibis, as I understood, `equals` usually returns a boolean, not an ibis expression.

I also updated the documentation about `.equals`

File: ibis/backends/postgres/tests/test_postgis.py [hunk 2]
```diff
@@ -149,9 +149,16 @@ def test_geo_disjoint(geotable):
...
-    expr = geotable.geo_point.equals(geotable.geo_point)
+    # note: == operation just works for comparison between same datatypes
+    expr = geotable.geo_point == geotable.geo_point
     assert expr.execute().all()
 
+    expr = geotable.geo_point.geo_equals(geotable.geo_point)
+    assert expr.execute().all()
+
+    expr = geotable.geo_point.geo_equals(geotable.geo_linestring)
+    assert not expr.execute().all()
```
Comment (datapythonista) [inline ibis/backends/postgres/tests/test_postgis.py:160]: Do you want all them to be different? Then `any` should be better I think.

**Review by datapythonista — APPROVED** [review]
lgtm

File: ibis/backends/postgres/tests/test_postgis.py (hunk 2 above)
Comment (xmnlab) [reply ibis/backends/postgres/tests/test_postgis.py:160]: that makes sense. I will change it now. thanks @datapythonista

**Comment by jreback** [comment]
thanks @xmnlab

**Pull request merged**